wolfcrypt/src/signature.c

```diff
@@ -37,13 +39,43 @@
 
 /* If ECC and RSA are disabled then disable signature wrapper */
 #if !defined(HAVE_ECC) && defined(NO_RSA)
-#undef NO_SIG_WRAPPER
-#define NO_SIG_WRAPPER
+    #undef NO_SIG_WRAPPER
+    #define NO_SIG_WRAPPER
 #endif
 
 /* Signature wrapper disabled check */
 #ifndef NO_SIG_WRAPPER
 
+#if !defined(NO_RSA) && !defined(NO_ASN)
+static int wc_SignatureRsaEncode(enum wc_HashType hash_type, byte** hash_data,
+    word32* hash_len)
+{
+    int ret = wc_HashGetOID(hash_type);
+    if (ret > 0) {
+        int oid = ret;
+
+        /* Allocate buffer for hash and encoded ASN header */
+        word32 digest_len = *hash_len + MAX_ALGO_SZ;
```

Comment:
I think digest_len should also account for the length encoding and sequence header.

MAX_ENCIDED_DIG_SZ + MAX_ALGO_SZ + MAX_SEQ_SZ would cover it.